### ISSUE TITLE
Bind (spell) can break its own Bind (effect) - BindBreakCheck does not check if spell just cast on mob was actually Bind itself

### ISSUE THREAD

**Issue by xipies** [DarkstarProject/darkstar#2720]
Attack a mob (melee, ranged, nuke, enfeeble such as paralyze), then cast bind. Assuming bind lands, it appears to always wear immediately.

I was able to bind a mob, wait for bind to wear naturally, then rebind. But I have not been able to bind a mob after any sort of of "hit" against it.

Revision: e392f8d
Server: Classic

**Comment by helixhamin** [comment]
Do you mean that once you have hit it, if you stop doing damage, and then try to rebind, the new bind will not work, or do you mean that when you bind a monster and then hit it, bind will wear off?

From BG-Wiki:
Dealing direct damage to a bound mob can cause Bind to wear.

**Comment by xipies** [comment]
1. Hit mob once, and only once.
2. Stop all attacks on mob. No DoT on mob.
3. Cast bind.
4. Bind status lands.
5. Bind status wears immediately (mob basically doesn't stop moving at all).

Can repeat 3-5, appears to always have the same result.

**Comment by helixhamin** [comment]
Looking at old issues, I came across this one: https://github.com/DarkstarProject/darkstar/issues/1213
Is this one still acting this way? Damaging the mobs when bind is applied? Sorry, I cannot check as my pc is DOA right now.

**Comment by xipies** [comment]
I believe it is still happening (#1213).

I know I did see this last night:
1. Mob unclaimed.
2. Cast enfeeble on mob. I don't remember which ones it happened with.
3. Mob shows claimed and ~98% hp.

Casting any more enfeebles does not actually damage the mob. Like lichtl stated, it is most likely a recalculation and not actual damage.

#1213 also predated this (#2720) issue by several months at least.

**Comment by helixhamin** [comment]
Yes but I am wondering if the two are related since damage to a mob can break bind.

**Comment by xipies** [comment]
Maybe. I did one other test last night, though:

1. Mob unclaimed.
2. Cast bind on mob. Bind lands and sticks.
3. Cast poison II on mob. Lands and starts taking away HP. Not breaking bind with initial cast or DoT ticks.
4. Let bind wear naturally.
5. Cast bind again. Bind lands and actually sticks.

I only did this sequence once, so not sure if it was a fluke.

**Comment by teschnei** [comment]
I can't reproduce this. Damaging a mob and recasting bind works as it should

**Comment by xipies** [comment]
I just tested locally and put in some debug statements in battleutils.BindBreakCheck.

I tested with 75BLM/37WHM in The Boyahda Tree near Dragon's Aery entrance on spiders.

1. Logged in.
2. Cast bind.
3. PDefender->StatusEffectContainer->HasStatusEffect(EFFECT_BIND)
    Resolved to true
4. PDefender->StatusEffectContainer->HasStatusEffect(EFFECT_BIND)
    Also, resolved to true

Hence, Bind broke itself.

After many castings, I could see, I assume based on the the mobs conning T, that was why I saw seeing it break very often (75% chance in the code).
